### ISSUE TITLE
Tests get stuck on a docker container dedicated to build

### ISSUE THREAD

**Issue by sergeiqiniu** [jhipster/generator-jhipster#3788]
##### **Overview of the issue**
Hi,

I am a newbie to jHipster. I have successfully managed to run the sample application on my development machine (local installation). However when I retrieve it from my GitLab on my build machine ```./mvnw test``` halts before displaying the results

##### **Motivation for or Use Case** 

Before dwelving into actual software development I would like to setup my continuous integration infrastructure. I do not have full control over my projects. The project leader want me to build my app inside a docker container.

I did not encounter any problem running the sample application on my workstation. I have committed the application directory to our GitLab machine, keeping the .gitignore supplied. I have created a docker image dedicated to building (basically openjdk8, docker, docker compose. I also added node, bower and gulp to see if they helped. They didn't).  When I clone my application and run ```./mvnw test``` the process is stuck after the integration tests I think:

```
2016-06-28 14:31:46.096  INFO 332 --- [           main] c.m.myapp.service.UserServiceIntTest     : Started UserServiceIntTest in 139.441 seconds (JVM running for 156.218)
Hibernate: select authority0_.name as name1_0_0_ from jhi_authority authority0_ where authority0_.name=?
...
lots of Hibernate stuff
...
Hibernate: select authoritie0_.user_id as user_id1_4_0_, authoritie0_.authority_name as authorit2_5_0_, authority1_.name as name1_0_1_ from jhi_user_authority authoritie0_ inner join jhi_authority authority1_ on authoritie0_.authority_name=authority1_.name where authoritie0_.user_id=?
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.793 sec - in com.mycompany.myapp.web.rest.UserResourceIntTest
2016-06-28 14:32:16.336  INFO 332 --- [       Thread-7] c.m.myapp.config.CacheConfiguration      : Remove Cache Manager metrics
2016-06-28 14:32:16.415  INFO 332 --- [       Thread-7] c.m.myapp.config.CacheConfiguration      : Closing Cache Manager
```

Whereas on my dev machine I have a successful build which ends like this

```
2016-06-28 09:52:52.543  INFO 3772 --- [       Thread-7] c.m.myapp.config.CacheConfiguration      : Closing Cache Manager

Results :

Tests run: 29, Failures: 0, Errors: 0, Skipped: 0

[INFO]
[INFO] --- jacoco-maven-plugin:0.7.6.201602180812:report (post-unit-test) @ myapplication ---
[INFO] Analyzed bundle 'Myapplication' with 101 classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 47.293 s
[INFO] Finished at: 2016-06-28T09:52:55+02:00
[INFO] Final Memory: 46M/364M
[INFO] ------------------------------------------------------------------------
```

Did I miss something ?

Thanks

##### **JHipster Version(s)** 

3.4.2
##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder** 

```
{
  "generator-jhipster": {
    "jhipsterVersion": "3.4.2",
    "baseName": "myapplication",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": "no",
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "4cbe2ac4942fb1e219f3689e30e93908e718bd70",
    "useSass": false,
    "applicationType": "monolith",
    "testFrameworks": [
      "gatling"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}
```
##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory** 

n/a

##### **Browsers and Operating System** 

No browser.

My dev machine is W7 64bits. 

My build container runs 
Linux 72c1c45c43f5 4.2.0-27-generic #32~14.04.1-Ubuntu SMP Fri Jan 22 15:32:26 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

##### **Reproduce the error** 

see above

##### **Related issues** 

None

##### **Suggest a Fix** 

No suggestion

**Comment by PierreBesson** [comment]
I would suggest basing your container on our official Docker Image based on Ubuntu: https://hub.docker.com/r/jhipster/jhipster/
(ie. start the Dockerfile by `FROM jhipster/jhipster:v3.4.2`)

It was created to let people generate their apps in docker but I think it would be a good starting point for what you want to do. See https://jhipster.github.io/installation/ for how to use it

**Comment by sergeiqiniu** [comment]
Thanks for the advice

Indeed I took the problem in the opposite direction generating my test app on my server inside the container you have indicated to me. I have then tested pushing a modification from my dev machine. Works fine.

Don't what happened when I was starting from the workstation. I had strange zombie process steeling all the CPU on the server.

BTW I think that I have noticed that the app directory in the container is owned by root

Cheers

Sergei
